modules/tribe/src/main/java/org/elasticsearch/tribe/TribeService.java

```diff
@@ -218,17 +155,31 @@ public static Settings processSettings(Settings settings) {
 
     private final NamedWriteableRegistry namedWriteableRegistry;
 
-    public TribeService(Settings settings, Path configPath, ClusterService clusterService, final String tribeNodeId,
-                        NamedWriteableRegistry namedWriteableRegistry, BiFunction<Settings, Path, Node> clientNodeBuilder) {
+    @Inject
```

Comment:
I strongly believe we should not add any new  `@Inject` anywhere in the codebase. we need to find a different way to do this